components/navigation/Filter.js

```diff
@@ -1,31 +1,144 @@
+                      return;
+                    } else {
+                      const newData = {
+                        value: i[checks[j].unique],
+                        text: i[checks[j].unique],
+                      };
+                      lists[name].push(newData);
+                      sortFilter(lists[name]);
+                    }
+                  } else {
+                    if (lists[name].some((data) => data.value === i)) {
+                      return;
+                    }
+                    const newData = {
+                      value: i,
+                      text: i,
+                    };
+                    lists[name].push(newData);
+                    sortFilter(lists[name]);
+                  }
+                });
+              } else {
+                if (lists[name].some((data) => data.value === result)) {
+                  break;
+                } else {
+                  const newData = {
+                    value: result,
+                    text: result,
+                  };
+                  lists[name].push(newData);
+                  sortFilter(lists[name]);
+                }
+              }
+            }
+          }
         }
-    });
-    const onFilterApply = (value) => {
-        setState(value);
-        const jobs = data.filter((e) => e.category === value);
-        onFilter(jobs);
+      }
+      setFilters(lists);
     }
-    return (
-        <div className="mb-4">
-            <Select options={categories} 
-            onChange={onFilterApply} selected={state}
-            className='w-56 max-w-sm'
-             />
-        </div>
-    )
+  }, [route]);
+  useEffect(() => {
+    onFilterApply();
+  }, [query]);
+  const onFilterApply = () => {
+    let result = data;
+    if (query && Object.keys(query).length >= 1) {
+      for (const property in query) {
+      const res = result.filter((e) => {
+        if (e[property] === query[property]) {
+          return (e[property] = query[property]);
+        }
+        if (Array.isArray(e[property])) {
+          if (e[property].some((data) => data.name === query[property])) {
+            return e[property].some((data) => data.name === query[property]);
+          } else if (e[property].includes(query[property])) {
+            return e[property].includes(query[property]);
+          }
+        }
+        return;
+      });
+        result = res;
+      }
+    }
+    onFilter(result)
+  };
+  return checks.map((check) => {
+    let selected = `Filter by ${check.name}...`;
+    if (Object.keys(query).length) {
+      if (query[check.name]) {
+        selected = `${query[check.name]}`;
+      }
+    }
+      return (
+        <Select
+          key={check.name}
+          options={filters[check.name]}
+          onChange={(e) => {
+            const newQuery = route.query;
+            route.push({
+              query: {
+                ...newQuery,
+                [check.name]: e,
+              },
+            });
```

Comment:
You should use the shallow routing here (https://nextjs.org/docs/routing/shallow-routing) and if it doesn't work, try to push new query to routing (with shallow routing!) as string not object :)

Reply:
@magicmatatjahu It's working now

Reply:
@derberg @fmvilas please test it now 🙏